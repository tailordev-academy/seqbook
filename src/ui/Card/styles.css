.Card {
  text-align: center;
}

.Card-title {
  text-transform: uppercase;
}

.Card-value {
  font-size: 3em;
}
