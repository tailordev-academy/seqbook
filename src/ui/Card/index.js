import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

const Card = props =>
  <div className="Card panel panel-default">
    <div className="Card-title panel-heading">
      <span className="panel-title">
        {props.title}
      </span>
    </div>
    <div className="Card-value panel-body">
      {props.value}
      {props.unit || ''}
    </div>
  </div>;

Card.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  unit: PropTypes.string,
};

export default Card;
