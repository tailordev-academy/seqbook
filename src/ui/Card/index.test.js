import React from 'react';
import renderer from 'react-test-renderer';

import Card from './index';

it('renders a card', () => {
  const tree = renderer
    .create(<Card title="card title" value="card value" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('can display a unit', () => {
  const tree = renderer
    .create(<Card title="card title" value="10" unit="%" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('can take numbers as value', () => {
  const tree = renderer
    .create(<Card title="card title" value={10} unit="%" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
