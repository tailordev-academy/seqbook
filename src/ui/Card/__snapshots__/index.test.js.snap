// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can display a unit 1`] = `
<div
  className="Card panel panel-default"
>
  <div
    className="Card-title panel-heading"
  >
    <span
      className="panel-title"
    >
      card title
    </span>
  </div>
  <div
    className="Card-value panel-body"
  >
    10
    %
  </div>
</div>
`;

exports[`can take numbers as value 1`] = `
<div
  className="Card panel panel-default"
>
  <div
    className="Card-title panel-heading"
  >
    <span
      className="panel-title"
    >
      card title
    </span>
  </div>
  <div
    className="Card-value panel-body"
  >
    10
    %
  </div>
</div>
`;

exports[`renders a card 1`] = `
<div
  className="Card panel panel-default"
>
  <div
    className="Card-title panel-heading"
  >
    <span
      className="panel-title"
    >
      card title
    </span>
  </div>
  <div
    className="Card-value panel-body"
  >
    card value
    
  </div>
</div>
`;
